src/UmbMapper/UmbMapperConfig{T}.cs

```diff
@@ -210,29 +187,78 @@ object IUmbMapperConfig.Map(IPublishedContent content)
             // Certain mappers like Archtype require the context so we want to ensure it exists.
             EnsureUmbracoContext();
 
-            // Now map the non-lazy properties
+            // Now map the non-lazy properties and non-lazy predicate mappings
+            this.MapNonLazyProperties(content, result);
+
+            return result;
+        }
+
+        /// <inheritdoc/>
+        public void Map(IPublishedContent content, object destination)
+        {
+            // TODO: If we're mapping to an existing object, what do we do with lazy items?
```

Comment:
Ok, I think I know how to get around this...

We can:

1. Make a public type creation method. `UmbMapper.CreateEmpty<T>`. This can either create for us a normal instance or the proxy depending on whether the `UmbMapperConfig<T>` contains lazy properties. This would be transparent. (We may have to assign an empty interceptor here in case someone tries to set a virtual property outside of UmbMapper).
2. In `Map(IPublishedContent content, object destination)` we can check if the destination object implements `IProxy`. If it does, create the property dictionary as before and assign it to the the instances `Interceptor` property. To handle non proxies we simply create a copy of `MapNonLazyProperties(IPublishedContent content, object destination)` but for our lazy maps that will simply map the values as normal.

Does that make sense?

If you ticked the box to give me access to the PR I can push the changes to there, or even simply pull your current progress into another branch and continue to work onit.

Reply:
Yes, that sounds good to me.  I've ticked "allow edits from maintainers" but please go ahead however is easiest for you.  Thanks.

Reply:
@AndyButland Great, will be finished soon. 

Quick question. Would you expect any additional mapping to overwrite any properties you have already set?
e.g

``` c#
PublishedItem result = UmbMapperRegistry.CreateEmpty<PublishedItem>();

// Set a value before mapping.
result.Id= 1;

// Mapping pulls Id from IPublishedContent since it's been configured to do so and sets it in our destination to 2
content.MapTo(result);

Console.WriteLine(result.Id) // 2
```

Reply:
Hmm, good question.  On balance I think it probably should yes.  Just considering how something like AutoMapper would work, and pretty sure it overwrites in this situation.  And as the developer using the tool, you'd have some way to work around this if it's not the behaviour you wanted.

Reply:
Glad we agree, the performance hit would have been harsh. I've pushed updates if you want to cast your eye over them.